lib/lhm.rb

```diff
@@ -99,12 +99,13 @@ def setup(connection, connection_options = {})
   # @option connection_options [Boolean] :reconnect_with_consistent_host
   #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
   def connection(connection_options = nil)
-    if @@connection.nil?
+    @@connection ||= begin
```

Comment:
I truly believe this function should ONLY return a connection. However, I don't want to break anything for the downstream users... However, this is a change that should be done for the next major release to lhm-shopify 4

Reply:
This code is a bit complex and surprising as you have pointed out.. but I guess this is fine for now.